Assets/MRTK/SDK/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -27,15 +27,26 @@ public class MigrationTool
         /// </summary>
         public List<Type> MigrationHandlerTypes => new List<Type>(migrationHandlerTypes);
 
-        private List<Object> migrationObjects = new List<Object>();
+        private Dictionary<Object, MigrationStatus> migrationObjects = new Dictionary<Object, MigrationStatus>();
         /// <summary>
-        /// Returns a copy of all game objects, prefabs and scene assets selected for migration
+        /// Returns a copy of all game objects, prefabs and scene assets selected for migration and their migration status
         /// </summary>
-        public List<Object> MigrationObjects => new List<Object>(migrationObjects);
+        public Dictionary<Object, MigrationStatus> MigrationObjects => new Dictionary<Object, MigrationStatus>(migrationObjects);
 
         private IMigrationHandler migrationHandlerInstance;
         private Type migrationHandlerInstanceType;
 
+        /// <summary>
+        /// Possible states for the migration tool
+        /// </summary>
+        public enum MigrationState
+        {
+            PreMigration, // New object selection can be added to migration objects collection
+            Migrating, // Processing migration objects
+            PostMigration // New objects should not be added to migration objects collection
+        };
+        public MigrationState migrationState { get; private set; }
```

Comment:
Since this is a public property, the [name should be in PascalCase](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#naming-conventions) and have [a `<summary>` tag](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#function--method-summary-headers).